widget/editor.go

```diff
@@ -99,6 +112,39 @@ type maskReader struct {
 	overflow []byte
 }
 
+// combinedPos is a point in the editor, three different ways.
+type combinedPos struct {
+	// editorBuffer offset, from e.rr.caret. The other three fields are based
+	// off of this one.
+	ofs int
+
+	// lineCol.Y = line (offset into e.lines), and X = col (offset into
+	// e.lines[Y])
+	lineCol screenPos
+
+	// Pixel coordinates
+	x fixed.Int26_6
+	y int
+}
+
+type SelectionAction int
+
+const (
+	SelectionDoNotChange SelectionAction = iota
+	SelectionClear
+	SelectionExtend
+)
+
+type EditorOption struct {
```

Comment:
Is EditorOption overkill? What other options do you anticipate?

Reply:
I don't anticipate any other options.  It does seem a little overkill for that one option, I do agree, and moreover thought that already.

On the other hand, I was leery of changing the public API for Editor.Move, and I really didn't want to make the selection action a required argument, and a functional argument seemed like the best compromise.  Also I thought if we ever decided to export any of the other moveXXX functions (moveLines, etc), which seems possible, then the scaffolding would already be there to let them accept functional arguments too.

I think performance-wise the impact is minimal.

Reply:
Still, EditorOption seems overkill (in API, not in performance loss) just to avoid an argument. A specialized method seems a better choice.